File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        int l=nums.length;
        long[] prefixSum=new long[l];
        long[] suffixSum=new long[l];
        long prefix=0,suffix=0;
        for(int i=0;i<l;i++){
            prefix+=nums[i];
            prefixSum[i]=prefix;
        }
        for(int i=l-1;i>-1;i--){
            suffix+=nums[i];
            suffixSum[i]=suffix;
        }
        int count=0;
        for(int i=0;i<l-1;i++){
            if(prefixSum[i]>=suffixSum[i+1]){
                count++;
            }
        }
        return count;
    }
}

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        prefixSum=[0]*len(nums)
        suffixSum=[0]*len(nums)
        prefix=suffix=0
        for i in range(len(nums)):
            prefix+=nums[i]
            prefixSum[i]=prefix
        for i in range(len(nums)-1,-1,-1):
            suffix+=nums[i]
            suffixSum[i]=suffix
        #print(prefixSum,suffixSum)
        count=0
        for i in range(len(nums)-1):
            if prefixSum[i]>=suffixSum[i+1]:
                count+=1
        return count